Sources/Datadog/RUM/Instrumentation/Views/SwiftUI/SwiftUIRUMViewsHandler.swift

```diff
@@ -0,0 +1,195 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+import UIKit
+
+/// Publisher generating RUM Commands on `SwiftUI.View` events.
+internal protocol SwiftUIViewHandler: RUMCommandPublisher {
+    /// Respond to a `SwiftUI.View.onAppear` event.
+    func onAppear(identity: String, name: String, path: String, attributes: [AttributeKey: AttributeValue])
+
+    /// Respond to a `SwiftUI.View.onDisappear` event.
+    func onDisappear(identity: String)
+}
+
+internal final class SwiftUIRUMViewsHandler: SwiftUIViewHandler {
+    /// RUM representation of a `SwiftUI.View`.
+    private struct View {
+        /// The RUM View identity.
+        let identity: String
+
+        /// View name used for RUM Explorer.
+        let name: String
+
+        /// View path used for RUM Explorer.
+        let path: String
+
+        /// Custom attributes to attach to the View.
+        let attributes: [AttributeKey: AttributeValue]
+    }
+
+    /// The current date provider.
+    private let dateProvider: DateProvider
+
+    /// The notification center where this handler observe the following notifications:
+    /// - `UIApplicationDidEnterBackgroundNotification`
+    /// - `UIApplicationWillEnterForegroundNotification`
+    private weak var notificationCenter: NotificationCenter?
+
+    /// The RUM Command subscriber responsible for processing
+    /// this publisher's commands.
+    private weak var subscriber: RUMCommandSubscriber?
+
+    /// The appearing views stack.
+    ///
+    /// This stack allows to track appearing and disappearing views to consistently
+    /// publish start and stop commands to the subscriber. The last item of the
+    /// stack is the visible one, any items below it have appeared before but not yet
+    /// disappeared. Therefore, they are considered not visible but can be revealed
+    /// if the last item disappears.
+    private var stack: [View] = []
```

Comment:
> With UIKit auto-instrumentation, this is done with the help of `UIKitHierarchyInspector` and a [simple state management](https://github.com/DataDog/dd-sdk-ios/blob/master/Sources/Datadog/RUM/AutoInstrumentation/Views/UIKitRUMViewsHandler.swift#L88L104).

Indeed, it's all true and very good observation 👌. Thanks for more insights - clearly, we can't handle value-based `SwiftUI.Views` in the same way as we do with object-based `UIKit.UIViewControllers`. The `stack` seems like a good, minimal approach 👍.

Speaking of `RUMM-1745`, I don't have any clear idea ATM. Like you say, having a single handler that will receive both `notify_viewDidAppear()` and `onAppear()` callbacks could be a solution. This way we could be considering both `UIKit` and `SwiftUI` navigation in one place, possibly leading to more control. Another approach could be to introduce next handler that combines `UIKit` and `SwiftUI` handlers into single stream - in theory that would mean less refactoring, but in practise it can be not doable if the hybrid support is mainly about details.